src/dscanner/analysis/incorrect_infinite_range.d

```diff
@@ -13,85 +13,73 @@ import dparse.lexer;
 /**
  * Checks for incorrect infinite range definitions
  */
-final class IncorrectInfiniteRangeCheck : BaseAnalyzer
+extern(C++) class IncorrectInfiniteRangeCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
 	mixin AnalyzerInfo!"incorrect_infinite_range_check";
 
 	///
-	this(string fileName, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, null, skipTests);
+		super(fileName);
 	}
 
-	override void visit(const StructBody structBody)
+	override void visit(AST.StructDeclaration sd)
 	{
 		inStruct++;
-		structBody.accept(this);
+		super.visit(sd);
 		inStruct--;
 	}
 
-	override void visit(const FunctionDeclaration fd)
+	override void visit(AST.ClassDeclaration cd)
 	{
-		if (inStruct > 0 && fd.name.text == "empty")
-		{
-			line = fd.name.line;
-			column = fd.name.column;
-			fd.accept(this);
-		}
+		inStruct++;
+		super.visit(cd);
+		inStruct--;
 	}
 
-	override void visit(const FunctionBody fb)
+	override void visit(AST.FuncDeclaration fd)
 	{
-		if (fb.specifiedFunctionBody && fb.specifiedFunctionBody.blockStatement !is null)
-			visit(fb.specifiedFunctionBody.blockStatement);
-		else if (fb.shortenedFunctionBody && fb.shortenedFunctionBody.expression !is null)
-			visitReturnExpression(fb.shortenedFunctionBody.expression);
-	}
+		import dmd.astenums : Tbool;
 
-	override void visit(const BlockStatement bs)
-	{
-		if (bs.declarationsAndStatements is null)
-			return;
-		if (bs.declarationsAndStatements.declarationsAndStatements is null)
-			return;
-		if (bs.declarationsAndStatements.declarationsAndStatements.length != 1)
-			return;
-		visit(bs.declarationsAndStatements);
-	}
+		AST.TypeFunction tf = fd.type.isTypeFunction();
+		AST.ReturnStatement rs = fd.fbody ? fd.fbody.isReturnStatement() : null;
+		AST.CompoundStatement cs = fd.fbody ? fd.fbody.isCompoundStatement() : null;
 
-	override void visit(const ReturnStatement rs)
-	{
-		if (inStruct == 0 || line == size_t.max) // not within a struct yet
-			return;
-		visitReturnExpression(rs.expression);
-	}
+		if (fd.ident && fd.ident.toString() == "empty" && tf && tf.next && tf.next.ty == Tbool && inStruct)
```

Comment:
This code is ugly, you should just use early return. E.G:

```d
if (!tf || !tf.next || tf.next.ty != bool)
    return;

if (!fd.ident || fd.ident.toString() != "empty")
   return;

....

```

It looks much cleaner and it avoid this ridiculous indentation level.